### PR TITLE
feat: argo sso groups field change to configurable param.

### PR DESCRIPTION
**Pull request by Shanjian2033**
Most of the explanation for this change is in https://github.com/argoproj/argo-workflows/issues/11034.
Code Change For: make argo sso groups field change to configurable param.

### CODE REVIEW

**Review by Shanjian2033 — COMMENTED** [review]
@alexec Can you help review and approve the pull request? Thanks!

**Comment by juliev0** [comment]
Sorry for the delay. I'm reading https://github.com/argoproj/argo-workflows/issues/11034. You wrote `Because the user info got from remote userInfoPath does not have "groups" field, but "memberof" field`. Do you have information on the difference between the two fields? (any link to documentation?)

File: server/auth/types/claims_test.go [hunk 1]
```diff
@@ -222,7 +222,7 @@ func TestGetUserInfoGroups(t *testing.T) {
 		httpClient = &HttpClientMock{StatusCode: 200, Body: body}
 
 		claims := &Claims{}
-		groups, err := claims.GetUserInfoGroups("Bearer fake", "https://fake.okta.com", "/user-info")
+		groups, err := claims.GetUserInfoGroups("Bearer fake", "https://fake.okta.com", "/user-info", "")
 		assert.Equal(t, groups, []string{"Everyone"})
```
Comment (juliev0) [inline server/auth/types/claims_test.go:226]: how much work to add a unit test for your case as well?

**Comment by agilgur5** [comment]
You can already use `customGroupClaimName` to achieve this. See https://argoproj.github.io/argo-workflows/argo-server-sso/#custom-claims

**Pull request closed without merging**